### DISCUSSION TITLE
502: Bad Gateway error

### DISCUSSION THREAD

**Discussion by pablosuarez999** [Koenkk/zigbee2mqtt#12476] [Devices]
Hi everyone
First of all, I have never programmed, so I understand really few things.
I have a problem with all addons, but zigbee2mqtt. I have installed Home Assistant Supervised, and after creating an user for mqtt browser, and putting it in zigbee2mqtt's configuration, when I'm trying to open the web interface, I have a message that says: 502: Bad Gateway. 
I guess I have a configuration problem, please if someone could help me...

**Comment by pablosuarez999** [comment]
This is my options configuration: 

data_path: /config/zigbee2mqtt
socat:
  enabled: false
  master: pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777
  slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
  options: '-d -d'
  log: false
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://localhost
  user: mqtt_usuario
  password: xxxx
serial:
  port: >-
    /dev/serial/by-id/usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_88d278b00c13ec119df023c7bd930c07

**Reply by DaywaIker313** [reply]
data_path: /config/zigbee2mqtt
external_converters: []
devices: devices.yaml
groups: groups.yaml
homeassistant: true
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
serial:
  port: /dev/ttyUSB0
  user: mqtt_user1
  password: XXX
advanced:
  log_level: warn
  pan_id: 6754
  channel: 11
  network_key:
    - 1
    - 3
    - 5
    - 7
    - 9
    - 11
    - 13
    - 15
    - 0
    - 2
    - 4
    - 6
    - 8
    - 10
    - 12
    - 13
  availability_blocklist: []
  availability_passlist: []
device_options: {}
blocklist: []
passlist: []
queue: {}
frontend:
  port: 8099
experimental: {}
socat:
  enabled: false
  master: pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777
  slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
  options: '-d -d'
  log: false

I think your port is wrong.

**Comment by pablosuarez999** [comment]
I have just copied your config but changed my user and password and I have the same problem. This is the log:

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] socat.sh: executing... 
[14:20:36] INFO: Socat not enabled, marking service as down
[cont-init.d] socat.sh: exited 0.
[cont-init.d] zigbee2mqtt.sh: executing... 
[cont-init.d] zigbee2mqtt.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[14:20:40] INFO: Handing over control to Zigbee2mqtt Core ... 
> zigbee2mqtt@1.25.1 start
> node index.js
Zigbee2MQTT:info  2022-05-15 14:20:51: Logging to console and directory: '/config/zigbee2mqtt/log/2022-05-15.14-20-46' filename: log.txt
Zigbee2MQTT:info  2022-05-15 14:20:51: Starting Zigbee2MQTT version 1.25.1 (commit #unknown)
Zigbee2MQTT:info  2022-05-15 14:20:51: Starting zigbee-herdsman (0.14.27)

**Comment by DaywaIker313** [comment]
I think all you need to do is change your path to the correct one, in my case it is dev/ttyUSB0

![image](https://user-images.githubusercontent.com/41849953/168472809-c1183ea5-e4b5-45f4-9820-0b95cdce8ea3.png)

if you copy and paste and it is not in the correct line or position in yamal it will not work. 

https://www.youtube.com/watch?v=-YeObwEZqL4&t=577s

This is a good video, did help me to get it sorted, had the same issue as you before. Also, after you restart it will take a moment till it will load.

Hope that helps

**Reply by pablosuarez999** [reply]
I checked hardware and tutorials and its correct like you, I tried that last one because of a video
